clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -90,121 +90,6 @@ def CIR_BoolType :
   let hasCustomAssemblyFormat = 1;
 }
 
-//===----------------------------------------------------------------------===//
-// StructType
-//
-// The base type for all RecordDecls.
-//
-//===----------------------------------------------------------------------===//
-
-def CIR_StructType : CIR_Type<"Struct", "struct",
-    [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
-
-  let summary = "CIR struct type";
-  let description = [{
-    Each unique clang::RecordDecl is mapped to a `cir.struct` and any object in
-    C/C++ that has a struct type will have a `cir.struct` in CIR.
-  }];
```

Comment:
I'm pretty sure there's a C++ way to convey the docs, can you please make sure it happens?

Reply:
> I'm pretty sure there's a C++ way to convey the docs

Any ideas where I can find this?

I couldn't find a way to integrate Tablegen and C++ docs. I can add Doxygen documentation to the new C++ `StructType` class, but that would not automatically integrate with the pretty Tablegen Markdown docs.

It seems that some [dialects manually maintain their docs](https://github.com/llvm/llvm-project/tree/main/mlir/docs/Dialects). I believe the solution here would be to change the way we update the docs: generate the Tablegen Markdown docs and then add any non-Tablegen documentation that is missing.

Reply:
> Any ideas where I can find this?

I was assuming some of that get converted to C++, but it's possible it's just emitted to markdown directly by tablegen, I haven't checked though.

> I believe the solution here would be to change the way we update the docs: generate the Tablegen Markdown docs and then add any non-Tablegen documentation that is missing.

That would probably work, though I imagine if it's worth it. I'm happy with some regular version in the comments, as long as the content is there as a reference. After this lands, we can think about replicating that directly in the webpage, if anything.